### PR TITLE
Introduce sql-server module

### PR DESCRIPTION
**Pull request by petersin0422**

### CODE REVIEW

File: azure/sql-server/README.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+        The max size of the database in gigabytes.        
+
+    - (object) **`dtu_model = null`** _[since v0.0.1]_
+
+        Configures the database using the DTU pricing model
+
+        - (string) **`tier`** _[since v0.0.1]_
+
+            Defines the tier of this database. Possible values are `"Basic"`, `"Standard"`, and `"Premium"`. Note that some tiers are not available for some regions. Run this CLI command to get a list of tiers applicable to your region. `az sql db list-editions --location westus --output table`
```
Comment (allejo) [inline azure/sql-server/README.md]: From reading this, I'm to assume that `--location westus` needs to be changed based on a person's need? If so, I'll rephrase this sentence a bit.

File: azure/sql-server/README.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+        Whether or not this database is zone redundant, which means the replicas of this database will be spread across multiple availability zones. This property is only settable for `Premium` and `BusinessCritical` tiers.
+
+- (object) **`firewall = null`** _[since v0.0.1]_
+
+    Manages firewall rules to allow incoming traffic
+
+    - (map(string)) **`rules`** _[since v0.0.1]_
+
+        A map of firewall rules in the following format `{rule_name = start_ip - end_ip}`. For example. `"{Office's Network" = "1.2.3.4 - 5.6.7.8"}`. If `start_ip` and `end_ip` are identical, you can omit `end_ip`. For example. `"{Peter's home network" = "1.2.3.4"}`
```
Comment (allejo) [inline azure/sql-server/README.md:206]:
```suggestion
        A map of firewall rules in the following format: `{"rule_name" = "start_ip - end_ip"}`. For example. `{"Office's Network" = "1.2.3.4 - 5.6.7.8"}`. If `start_ip` and `end_ip` are identical, you can omit `end_ip`. For example. `{"Peter's home network" = "1.2.3.4"}`
```

I think this is what you mean in your examples?

File: azure/sql-server/README.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+    Defines the administrator login credential for this SQL server, can be used with `azuread_authentication`
+
+    - (string) **`admin_username`** _[since v0.0.1]_
+
+        Username of the admin account
+
+    - (string) **`admin_password`** _[since v0.0.1]_
+
+        Password of the admin account
```
Comment (allejo) [inline azure/sql-server/README.md:234]:
```suggestion
        Password of the admin account in plain text
```

Might be useful to specify that it's in plain text? Or in some other format

**Review by allejo — CHANGES_REQUESTED** [review]

File: azure/sql-server/variables.tf [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  default     = {}
+}
+
+variable "additional_tags_all" {
+  type        = map(string)
+  description = "Additional tags for all resources in deployed with this module"
+  default     = {}
+}
+
+variable "azuread_authentication" {
```
[the remark this answers is not in the document]
Comment (petersin0422) [reply azure/sql-server/variables.tf:27]: sounds good

File: azure/sql-server/README.md (hunk 1 above)
Comment (petersin0422) [reply azure/sql-server/README.md]: clarification added.

File: azure/sql-server/README.md (hunk 2 above)
Comment (petersin0422) [reply azure/sql-server/README.md:206]: good catch. fixed

**Review by allejo — APPROVED** [review]

**Pull request merged**